### PR TITLE
Added device info support

### LINKED ISSUE #5: Define deviceInfo (hardware) taxonomy
Related to: https://github.com/CycloneDX/specification/issues/36

### PR DESCRIPTION
**Pull request by stevespringett**
This resolves #5 and supersedes https://github.com/CycloneDX/specification/issues/36

### CODE REVIEW

File: cdx.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 | Namespace | Description | Administered By | Taxonomy |
 | --- | --- | --- | --- |
+| `cdx:deviceInfo` | Namespace for properties specific to hardware devices. | CycloneDX Core Working Group | [cdx:deviceInfo taxonomy](cdx/deviceInfo.md) |
```
Comment (coderpatros) [inline cdx.md:5]: Namespaces should currently be lowercase

File: cdx/deviceInfo.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+## `cdx:deviceInfo` Namespace Taxonomy
+
+| Namespace             | Description                                                       |
+| --------------------- | ----------------------------------------------------------------- |
+| `cdx:deviceInfo:quantity`  | The total number of the specified component. |
+| `cdx:deviceInfo:function`  | The purpose of the component (Bluetooth, network, storage, microprocessor, connector, etc)                         |
```
Comment (coderpatros) [inline cdx/deviceInfo.md:6]: I think we should be more explicit about what values can be used for function

File: cdx/deviceInfo.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+| Property                            | Description                  |
+| ----------------------------------- | ---------------------------- |
+| `cdx:deviceInfo:bom:ebom`  | Location to the Engineering Bill of Materials. This BOM contains assembly-component structure with documents coming from the CAD systems along with information about how the product is engineered. |
+| `cdx:deviceInfo:bom:mbom`  | Location to the Manufacturing Bill of Materials. MBOM represents the data that is needed to perform product assembly. |
+
+## `cdx:deviceInfo:certifications` Namespace Taxonomy
+
+| Property                            | Description                  |
+| ----------------------------------- | ---------------------------- |
+| `cdx:deviceInfo:certifications:<AUTHORITY>`  | A certifying authority (e.g. FCC). |
```
Comment (coderpatros) [inline cdx/deviceInfo.md:26]: Same comment about being explicit about what values should be used for authority

File: cdx.md (hunk 1 above)
Comment (stevespringett) [reply cdx.md]: Thank you. I renamed deviceInfo to simply: device

File: cdx/deviceInfo.md (hunk 2 above)
Comment (stevespringett) [reply cdx/deviceInfo.md]: Being explicit defeats the purpose of properties, as we'd have to define a taxonomy that included everything from case assemblies, terminals, screws, resistors, fuzes, and of course all the different types of chips incluidng those used for communication and processing. And if we used different words than a designer or manufacturer did with an eBOM or mBOM, then we'd have to provide guidance there as well. I'd rather have this used however to match how the designer or manufacturer defines the function of something.

File: cdx/deviceInfo.md (hunk 3 above)
Comment (stevespringett) [reply cdx/deviceInfo.md]: I think this should be the abbreviation of the certifying agency. However, I don't think its practical to come up with a list of all of them.

Comment (stevespringett) [reply cdx/deviceInfo.md]: I do think we can provide some guidance however and include examples such as: UL, CE, EMC, FCC and CSA.

File: cdx/deviceInfo.md (hunk 2 above)
Comment (coderpatros) [reply cdx/deviceInfo.md]: Sounds good to me.

File: cdx/deviceInfo.md (hunk 3 above)
Comment (coderpatros) [reply cdx/deviceInfo.md]: If it's the abbreviation should we worry about collisions across countries?

Comment (coderpatros) [reply cdx/deviceInfo.md]: Maybe another level to the country code, i.e. `cdx:deviceinfo:certifications:us:fcc`?

Comment (stevespringett) [reply cdx/deviceInfo.md]: i like that. I've changed this to use ISO-3166 Alpha-2 country codes.

`cdx:device:certifications:<ISO-3166-1>:<AUTHORITY>`

ISO-3166 however, specifically states the codes are upper case.

**Review by coderpatros — APPROVED** [review]
I like this. I think merge it in.

Maybe we should announce it on the mailing list to attract interest/feedback from folks who work for device manufacturers?

File: cdx/deviceInfo.md (hunk 3 above)
Comment (coderpatros) [reply cdx/deviceInfo.md]: Upper case isn't a showstopper. They property namespaces _should_ be lower case. But I think this is a valid case not to follow that.

**Comment by stevespringett** [comment]
Go for it - and the announcement

**Pull request merged**